types/application.go

```diff
@@ -6,25 +6,42 @@ import (
 
 // "Application" is the base structure for a Pocket Network Application
 type Application struct {
-	Account         `json:"routing"`
-	RequestedChains []ApplicationRequestedBlockchain `json:"requestedChains"`
+	Account               `json:"routing"`
+	LegacyRequestedChains []LegacyApplicationRequestedBlockchain `json:"legacyRequestedChains"` // todo phase out
+	RequestedBlockchains  ApplicationRequestedBlockchains        `json:"requestedChains"`
 }
 
 type Applications []Application
 
-func NewApplication(address Address, publicKey AccountPublicKey, balance POKT, stakeAmount POKT, requestedChains []ApplicationRequestedBlockchain) Application {
+func NewApplication(address Address, publicKey AccountPublicKey, balance POKT, stakeAmount POKT, requestedChains ApplicationRequestedBlockchains) Application {
 	return Application{
 		Account: Account{
 			Address:     address,
 			PubKey:      publicKey,
 			Balance:     balance,
 			StakeAmount: stakeAmount,
 		},
-		RequestedChains: requestedChains,
+		RequestedBlockchains: requestedChains,
 	}
 }
 
-type ApplicationRequestedBlockchain struct {
+type LegacyApplicationRequestedBlockchain struct {
 	legacy.Blockchain    `json:"blockchain"`
 	AllocationPercentage uint8 `json:"allocationPercentage"`
 }
+
+type ApplicationRequestedBlockchains []ApplicationRequestedBlockchain
+
+type ApplicationRequestedBlockchain struct {
+	Blockchain           `json:"blockchain"`
+	AllocationPercentage uint8 `json:"allocationPercentage"`
```

Comment:
So nodes have: HostedBlockchains{} which you are familiar with

Similarly, apps have: RequestedBlockchains{} 
-Aka non native chains they are requesting
-Allocation percentage is the fraction they want to allocate towards each blockchain.

Reply:
I believe this concept might be outdated, I thought ever since Mono-sessions were stablished you would get the same throughput for each blockchain you allotted traffic for.

Reply:
Why would we do that level of inflexibility? you mean to say, that if you stake 100 pocket for bitcoin and ethereum: 50 will go to bitcoin and 50 will go to ethereum?